lib/rule-list.js

```diff
@@ -155,17 +155,95 @@ export const SIMPLIFY_FRACTION = defineRuleString(
 // e.g. 2/-3 -> -2/3
 export const SIMPLIFY_SIGNS = defineRuleString('#a / -#b', '-#a / #b')
 
+// e.g. 2x/3 -> 2/3 x
+export const REWRITE_FRACTIONAL_POLYNOMIAL = defineRuleString(
+    '#a #b / #c', '#a / #c #b', {a: query.isNumber, b: isPolynomialTerm, c: query.isNumber}
+)
 
 // ADDING FRACTIONS
 
+// e.g. 2/5 + 2/5 -> (2 + 2)/5
 export const COMBINE_NUMERATORS =
     defineRuleString('#a_0 / #b + ...', '(#a_0 + ...) / #b')
 
-export const COMMON_DENOMINATOR =
-    defineRuleString(
-        '#a_0 / #b_0 + ...',
-        '(#a_0 * #eval(lcm(#b_0, ...) / #b_0)) / (#b_0 * #eval(lcm(#b_0, ...) / #b_0)) + ...'
-    )
+// Finds common denominator in any scenario
+
+// e.g. 2/6 + 1/4 -> (2 * 2) / (6 * 2) + (1 * 3) / (4 * 3)
+// e.g. 2/(3 + x) + 2/3 -> (2 * 3)/ (3 * (3 + x)) + 2 * (3 + x) / (3 * (3 + x))
+export const COMMON_DENOMINATOR = defineRule(
+    (node) => {
+        if (query.isAdd(node)) {
+            const terms = node.args
+            // checks if at least one of the terms is a fraction
+            const hasFraction = terms.some(term => query.isDiv(term))
+
+            let sameDenom
+            // rule cannot apply if all of the terms are fractions
+            // and they all have the same denominator
+            if (terms.every(term => query.isDiv(term))) {
```

Comment:
I think you should be able to run `matchNode` on the current node with the `#a_0 / #b_0 + ...` as the pattern to get the start/end indexes of the first group of 2 or more consecutive fractions.  You'll have the pattern to a matching function before you can use it with `matchNode`.